code/modules/reagents/reagent_containers/food/drinks/juicebox.dm

```diff
@@ -1,11 +1,63 @@
 /obj/item/chems/food/drinks/juicebox
+	icon = 'icons/obj/juicebox.dmi'
+	icon_state = "juicebox_base"
+	blend_mode = BLEND_OVERLAY
 	name = "juicebox"
+	gender = PLURAL
 	desc = "A small cardboard juicebox. Cheap and flimsy."
 	volume = 30
 	amount_per_transfer_from_this = 5
 	atom_flags = 0
 	matter = list(MATERIAL_CARDBOARD = 30)
 
+	var/juiceIcon = 'icons/obj/juicebox.dmi'
+	var/primaryColor = "#ff00000"
+	var/secondaryColor = null
+	var/logoColor = null
+	var/boxStyle = "basic"
+
+/obj/item/chems/food/drinks/juicebox/proc/set_colors(
+	primary,
+	secondary=null,
+	logo=null,
+	style="basic"
+)
+	primaryColor = primary
+	secondaryColor = secondary
+	logoColor = logo
+	boxStyle = style
+	return update_juicebox()
+
+/obj/item/chems/food/drinks/juicebox/on_update_icon()
+	appearance = update_juicebox()
+
+/obj/item/chems/food/drinks/juicebox/proc/update_juicebox()
+	var/mutable_appearance/newAppearance = new(src)
+
+	newAppearance.icon = juiceIcon
+	newAppearance.appearance_flags = RESET_COLOR
+	newAppearance.color = primaryColor
+
+	if (secondaryColor != null)
```

Comment:
General standard is `if(!isnull(foo))` if `FALSE` and `0` need to be checked for, and just straight `if(foo)` for if truthy/falsey (TRUE/1/not null, FALSE/0/null) is acceptable.